pkg/controller/disruption/disruption.go

```diff
@@ -667,17 +675,30 @@ func (dc *DisruptionController) getExpectedScale(pdb *policy.PodDisruptionBudget
 			}
 			if controllerNScale != nil {
 				controllerScale[controllerNScale.UID] = controllerNScale.scale
+				coveredPods = append(coveredPods, pod)
 				foundController = true
 				break
 			}
 		}
+		// If the pod has a controllerRef but no matching controller, it means that
+		// either the controller resource doesn't exist (i.e. we have  an orphaned pod)
+		// or the controller does not support the scale subresource. If this happens,
```

Comment:
I think there are three cases:

1. error: controller ownerRef for a group/kind we cannot resolve
2. error: controller ownerRef that implements /scale but for which a call to /scale failed
3. non-error?: controller ownerRef we can resolve but which does not implement /scale (we can tell this from discovery docs)

for purposes of determining scale, I would have expected case 3 to be treated like an uncontrolled pod

Reply:
So number 3 is currently being treated as an uncontrolled pod, but this is considered an error for `maxUnavailable` and `minAvailable` with percentage. If we have `minAvailable` with a number, then the controller doesn't call `getExpectedScale` so we never look at the ownerRef.